src/Microsoft.AspNet.Mvc.TagHelpers/Internal/GlobbingUrlBuilder.cs

```diff
@@ -137,6 +138,59 @@ private string ResolveMatchedPath(string matchedPath)
             return RequestPathBase.Add(relativePath).ToString();
         }
 
+        private class PathComparer : IComparer<string>
+        {
+            public int Compare(string x, string y)
+            {
+                // < 0 = x < y
+                // > 0 = x > y
+
+                if (x.Equals(y))
```

Comment:
I don't see why not. The `Matcher` will give you results for both cases if you're on a system that's case sensitive and you have files with different cases, so why not sort them accordingly?